config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_sqlinstances.sql.cnrm.cloud.google.com.yaml

```diff
@@ -169,20 +181,10 @@ spec:
                     type: boolean
                   masterHeartbeatPeriod:
                     description: Immutable. Time in ms between replication heartbeats.
+                    format: int64
                     type: integer
                   password:
                     description: Immutable. Password for the replication connection.
-                    oneOf:
```

Comment:
Personally I would prefer to not allow the user to specify the value in the resource CR directly. Instead, for secrets, the value should always come from a secret CR. Do you think it would be ok to make that change, even though it's backwards incompatible?

Reply:
I agree it is a best practice to always use the secret, but since we've been supporting plaintext value for so long and SQLInstance is one of the really important resources (i.e. if something wrong/backwards incompatible happens to it, we will get many extremely panicking customers), I think blocking the usage of `[secretField].value` in this resource at the moment is _not_ a good idea.

I'm okay with start doing it in any new CRDs, or in any less used / less important existing CRDs.

Reply:
Added some validation to ensure that both the password value and secret reference are not both specified at the same time.